src/fwrite.c

```diff
@@ -577,16 +581,14 @@ int compressbuff(z_stream *stream, void* dest, size_t *destLen, const void* sour
   stream->avail_out = *destLen;
   stream->next_in = (Bytef *)source; // don't use z_const anywhere; #3939
   stream->avail_in = sourceLen;
-  int err = deflate(stream, Z_FINISH);
-  if (err == Z_OK) {
-    // with Z_FINISH, deflate must return Z_STREAM_END if correct, otherwise it's an error and we shouldn't return Z_OK (0)
-    err = -9;  // # nocov
-  }
-  *destLen = stream->total_out;
-  return err == Z_STREAM_END ? Z_OK : err;
+  int err = deflate(stream, Z_SYNC_FLUSH);
+  *destLen = *destLen - stream->avail_out;
+  // *destLen = stream->total_out;
+  return err != Z_STREAM_ERROR ? Z_OK : err;
```

Comment:
Thanks for 5687a0ce45e7e4051a1b1a06ebde6bc708f0024b. I think the corresponding change here is:

```suggestion
  return (err != Z_STREAM_ERROR) ? Z_OK : err;
```